kuksa_databroker/databroker/src/grpc/kuksa_val_v1/conversions.rs

```diff
@@ -302,12 +302,13 @@ impl TryFrom<&proto::Field> for broker::Field {
 impl From<proto::Datapoint> for broker::Datapoint {
     fn from(from: proto::Datapoint) -> Self {
         Self {
-            ts: match from.timestamp {
+            ts: SystemTime::now(),
+            source_ts: match from.timestamp {
                 Some(ts) => match std::convert::TryInto::try_into(ts) {
-                    Ok(ts) => ts,
-                    Err(_) => SystemTime::now(),
+                    Ok(ts) => Some(ts),
```

Comment:
If this conversion fails, I don't think it makes sense to use `SystemTime::now()`. That doesn't reflect reality, as that was not what the source provided.

Unless we want to rework this conversion and return an error if the conversion fails (which would be the most truthful), I would just set it to `None` instead of `SystemTime::now()`.